instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/trace/SpanConfigE2ETests.kt

```diff
@@ -0,0 +1,239 @@
+     */
+    @Test
+    fun trace_config_feature_enabled() {
+        val testMethodName = "trace_config_feature_enabled"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                config = Configuration.Builder(
+                    logsEnabled = true,
+                    tracesEnabled = true,
+                    rumEnabled = true,
+                    crashReportsEnabled = true
+                ).build()
+            )
+        }
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: Credentials#constructor(String, String, String, String?, String? = null)
+     */
+    @Test
+    fun trace_config_feature_disabled() {
+        val testMethodName = "trace_config_feature_disabled"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                config = Configuration.Builder(
+                    logsEnabled = true,
+                    tracesEnabled = false,
+                    rumEnabled = true,
+                    crashReportsEnabled = true
+                ).build()
+            )
+        }
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: Configuration#Builder#fun setSpanEventMapper(com.datadog.android.event.SpanEventMapper): Builder
+     */
+    @Test
+    fun trace_config_set_span_event_mapper() {
+        val testMethodName = "trace_config_set_span_event_mapper"
+        val fakeResourceName = "truly-random"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                config = Configuration.Builder(
+                    logsEnabled = true,
+                    tracesEnabled = true,
+                    rumEnabled = true,
+                    crashReportsEnabled = true
+                ).setSpanEventMapper(
+                    object : SpanEventMapper {
+                        override fun map(event: SpanEvent): SpanEvent {
+                            if (event.resource == fakeResourceName) {
+                                event.name = testMethodName
+                                event.resource = testMethodName
+                            }
+                            return event
+                        }
+                    }
+                ).build()
+            )
+        }
+        GlobalTracer.get()
+            .buildSpan(fakeResourceName)
+            .start()
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: AndroidTracer#Builder#fun setBundleWithRumEnabled(Boolean): Builder
+     */
+    @Test
+    fun trace_config_set_bundle_with_rum_enabled() {
```

Comment:
You are not able to check the propagation of the `session_id` or `view_id` also ? I think those are propagated in the headers but is there any way you could check these in the query ?

Reply:
good point. I added calls for `startView`/`stopView` manually, but on top of `application_id` I can get only `session_id`, `view_id` is not there. Do you know why?